src/Cache.php

```diff
@@ -55,38 +73,39 @@ public function __construct(\Psr\SimpleCache\CacheInterface $handler = null)
      * @param mixed $key the key to be normalized
      * @return string the generated cache key
      */
-    private function buildKey($key): string
+    private function normalizeKey($key): string
     {
-        if (\is_string($key)) {
-            return ctype_alnum($key) && mb_strlen($key, '8bit') <= 32 ? $key : md5($key);
+        if (!$this->keyNormalization) {
+            $normalizedKey = $key;
+        } elseif (\is_string($key)) {
+            $normalizedKey = ctype_alnum($key) && mb_strlen($key, '8bit') <= 32 ? $key : md5($key);
```

Comment:
Not sure these checks are faster then just `json_encode` everything.

Reply:
This logic was there before.